api/admin/controller/individual_admin_settings.py

```diff
@@ -98,6 +98,104 @@ def append_role(roles, role):
             individualAdmins=admins,
         )
 
+    def process_post_create_first_admin(self, email: str):
+        """Create the first admin in the system."""
+
+        # Passwords are always required, so check presence and validity up front.
+        password: Optional[str] = flask.request.form.get("password")
+        if not self.is_acceptable_password(password):
+            return self.unacceptable_password()
+
+        success = False
+        try:
+            admin, _ = get_one_or_create(self._db, Admin, email=email)
+            self.check_permissions(admin, settingUp=True)
+
+            # Update the roles, if requested.
+            roles = flask.request.form.get("roles")
```

Comment:
You'll see this code repeated a couple of times.

My feeling was that, due to how twisted the rest of the code is and given the critical nature of authenticating and creating admins, it was more important to have something straightforward and readable for each case rather than save a few lines and thread more "are we setting up", "are we creating a new admin" flags through the existing code.